File: output/GPT4o-Mini/xCodeEval_4000/C++/Python/cc71589cdb8b0c04028e87bc7f1cb2e7.py
import sys
import random
import time

INF = int(1e9)
INF64 = int(1e18)
MOD = int(1e9) + 7
EPS = 1e-9
PI = 3.141592653589793

lim = [0, 9, 99, 99, 3099, 13099, 113099, 1113099, 11113099, 111113099]
pr1 = ["", "198", "19", "2", "", "", "", "", "", ""]
pr2 = ["", "199", "20", "3", "1", "1", "1", "1", "1", "1"]

def read():
    s = input().strip()
    return s[4:], len(s[4:])

def solve(s, n):
    x = int(s)
    if x >= lim[n]:
        return pr1[n] + s
    else:
        return pr2[n] + s

def main():
    if len(sys.argv) > 1 and sys.argv[1] == "_DEBUG":
        with open("input.txt", "r") as f:
            input_data = f.read().strip().splitlines()
        input_iter = iter(input_data)
    else:
        input_iter = iter(sys.stdin)

    if len(sys.argv) > 1 and sys.argv[1] == "MULTITEST":
        tc = int(next(input_iter))
        for _ in range(tc):
            s, n = read()
            print(solve(s, n))
    else:
        while True:
            try:
                s, n = read()
                print(solve(s, n))
            except EOFError:
                break

if __name__ == "__main__":
    main()

File: output/GPT4o-Mini/xCodeEval_4000/C++/Go/cc71589cdb8b0c04028e87bc7f1cb2e7.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

const (
	N      = 1000 + 7
	INF    = int(1e9)
	INF64  = int64(1e18)
	MOD    = int(1e9) + 7
	EPS    = 1e-9
	PI     = 3.141592653589793
)

var lim = []int64{0, 9, 99, 99, 3099, 13099, 113099, 1113099, 11113099, 111113099}
var pr1 = []string{"", "198", "19", "2", "", "", "", "", "", ""}
var pr2 = []string{"", "199", "20", "3", "1", "1", "1", "1", "1", "1"}

var n int
var s string

func read() bool {
	_, err := fmt.Scan(&s)
	if err != nil {
		return false
	}
	s = s[4:]
	n = len(s)
	return true
}

func solve() string {
	x := int64(0)
	fmt.Sscan(s, &x)
	if x >= lim[n] {
		return pr1[n] + s
	}
	return pr2[n] + s
}

func main() {
	rand.Seed(time.Now().UnixNano())

	var tc int
	fmt.Scan(&tc)
	for i := 0; i < tc; i++ {
		if read() {
			fmt.Println(solve())
		}
	}
}
